Algorithm.Framework/Selection/OpenInterestFutureUniverseSelectionModel.cs

```diff
@@ -0,0 +1,138 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using QuantConnect.Algorithm.Framework.Selection;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Algorithm.Farmework.Selection
+{
+    /// <summary>
+    ///     Selects contracts in a futures universe, sorted by open interest.  This allows the selection to identifiy current
+    ///     active contract.
+    /// </summary>
+    public class OpenInterestFutureUniverseSelectionModel : FutureUniverseSelectionModel
+    {
+        private readonly int _chainContractsLookupLimit;
+        private readonly IAlgorithm _algorithm;
+        private readonly int _resultsLimit;
+        private readonly SecurityExchangeHours _exchangeHours;
+
+        /// <summary>
+        ///     Creates a new instance of <see cref="OpenInterestFutureUniverseSelectionModel" />
+        /// </summary>
+        /// <param name="algorithm">Algorithm</param>
+        /// <param name="futureChainSymbolSelector">Selects symbols from the provided future chain</param>
+        /// <param name="chainContractsLookupLimit">Limit on how many contracts to query for open interest</param>
+        /// <param name="resultsLimit">Limit on how many contracts will be part of the universe</param>
+        public OpenInterestFutureUniverseSelectionModel(IAlgorithm algorithm, Func<DateTime, IEnumerable<Symbol>> futureChainSymbolSelector, int chainContractsLookupLimit = 6,
+            int resultsLimit = 1) : base(TimeSpan.FromDays(1), futureChainSymbolSelector)
+        {
+            if (algorithm == null)
+            {
+                throw new ArgumentNullException(nameof(algorithm));
+            }
+
+            _algorithm = algorithm;
+            _resultsLimit = resultsLimit;
+            _chainContractsLookupLimit = chainContractsLookupLimit;
+            _exchangeHours = SecurityExchangeHours.AlwaysOpen(TimeZones.Utc);
+        }
+
+        /// <inheritdoc cref="FutureUniverseSelectionModel.Filter" />
+        protected override FutureFilterUniverse Filter(FutureFilterUniverse filter)
+        {
+            return filter.Contracts(FilterByOpenInterest(filter));
+        }
+
+        /// <summary>
+        ///     Filters a set of contracts based on open interest.
+        /// </summary>
+        /// <param name="contracts">Contracts to filter</param>
+        /// <returns>Filtered set</returns>
+        public IEnumerable<Symbol> FilterByOpenInterest(IEnumerable<Symbol> contracts)
+        {
+            return contracts.OrderBy(x => x.ID.Date).Take(_chainContractsLookupLimit).OrderByDescending(GetOpenInterest).ThenBy(x => x.ID.Date).Take(_resultsLimit);
+        }
+
+        private decimal GetOpenInterest(Symbol symbol)
+        {
+            var current = _algorithm.Time;
+            var previousBusinessDate = GetPreviousBusinessDay(_algorithm.Time, symbol);
```

Comment:
Doesn't this assume the algorithm time zone defaulting to `TimeZones.NewYork`? 
Maybe it's better to use `algorithm.UtcTime` instead?

Reply:
Addressed, as mentioned in lower conversations.